diff --git a/sdk_includes/public/build.h b/sdk_includes/public/build.h
index 8eb7396..db72646 100644
--- a/sdk_includes/public/build.h
+++ b/sdk_includes/public/build.h
@@ -137,6 +137,8 @@ Then you can use another oneliner to query all variables:
 		#define XASH_NSWITCH 1
 	#elif defined __vita__
 		#define XASH_PSVITA 1
+	#elif defined __EMSCRIPTEN__
+		#define XASH_EMSCRIPTEN 1
 	#elif defined __wasi__
 		#define XASH_WASI 1
 	#elif defined __sun__
diff --git a/sdk_includes/public/buildenums.h b/sdk_includes/public/buildenums.h
index 77de60c..bba2dc9 100644
--- a/sdk_includes/public/buildenums.h
+++ b/sdk_includes/public/buildenums.h
@@ -59,6 +59,7 @@ For more information, please refer to <http://unlicense.org/>
 #define PLATFORM_SUNOS      16
 #define PLATFORM_HURD       17
 #define PLATFORM_PSP        18
+#define PLATFORM_EMSCRIPTEN 20
 
 #if XASH_WIN32
 	#define XASH_PLATFORM PLATFORM_WIN32
@@ -86,6 +87,8 @@ For more information, please refer to <http://unlicense.org/>
 	#define XASH_PLATFORM PLATFORM_NSWITCH
 #elif XASH_PSVITA
 	#define XASH_PLATFORM PLATFORM_PSVITA
+#elif XASH_EMSCRIPTEN
+	#define XASH_PLATFORM PLATFORM_EMSCRIPTEN
 #elif XASH_WASI
 	#define XASH_PLATFORM PLATFORM_WASI
 #elif XASH_SUNOS
